common/test/acceptance/pages/lms/discussion.py

```diff
@@ -308,6 +308,35 @@ def get_num_displayed_threads(self):
     def element_exists(self, selector):
         return self.q(css=self._discussion_selector + " " + selector).present
 
+    def is_new_post_opened(self):
+        return self._find_within(".new-post-article").visible
+
+    def click_element(self, selector):
+        self.wait_for_element_presence(
+            "{discussion} {selector}".format(discussion=self._discussion_selector, selector=selector),
+            "{selector} is visible".format(selector=selector)
+        )
+        self._find_within(selector).click()
+
+    def click_cancel_new_post(self):
+        self.click_element(".cancel")
+        EmptyPromise(
+            lambda: not self.is_new_post_opened(),
+            "New post closed"
+        ).fulfill()
+
+    def click_new_post_button(self):
```

Comment:
@andy-armstrong has added some similar methods in https://github.com/edx/edx-platform/pull/6493. You should sync up with him to avoid merge conflicts.